src/generic/problem.cc

```diff
@@ -8307,13 +8307,125 @@ namespace oomph
    actions_after_change_in_global_parameter();
   }*/
 
+  //==================================================================
+  /// Solve the eigenproblem. Legacy version that returns real vectors which are
+  /// related in some solver-specific way to the real and imaginary parts
+  /// of the actual, usually complex eigenvalues.
+  /// At least n_eval eigenvalues are computed.
+  //==================================================================
+  void Problem::solve_eigenproblem_legacy(
+    const unsigned& n_eval,
+    Vector<std::complex<double>>& eigenvalue,
+    Vector<DoubleVector>& eigenvector,
+    const bool& steady)
+  {
+    // If the boolean flag is steady, then make all the timesteppers steady
+    // before solving the eigenproblem. This will "switch off" the
+    // time-derivative terms in the jacobian matrix
+    if (steady)
+    {
+      // Find out how many timesteppers there are
+      const unsigned n_time_steppers = ntime_stepper();
+
+      // Vector of bools to store the is_steady status of the various
+      // timesteppers when we came in here
+      std::vector<bool> was_steady(n_time_steppers);
+
+      // Loop over them all and make them (temporarily) static
+      for (unsigned i = 0; i < n_time_steppers; i++)
+      {
+        was_steady[i] = time_stepper_pt(i)->is_steady();
+        time_stepper_pt(i)->make_steady();
+      }
+
+      // Call the Eigenproblem for the eigensolver
+      Eigen_solver_pt->solve_eigenproblem_legacy(
+        this, n_eval, eigenvalue, eigenvector);
+
+      // Reset the is_steady status of all timesteppers that
+      // weren't already steady when we came in here and reset their
+      // weights
+      for (unsigned i = 0; i < n_time_steppers; i++)
+      {
+        if (!was_steady[i])
+        {
+          time_stepper_pt(i)->undo_make_steady();
+        }
+      }
+    }
+    // Otherwise if we don't want to make the problem steady, just
+    // assemble and solve the eigensystem
+    else
+    {
+      // Call the Eigenproblem for the eigensolver
+      Eigen_solver_pt->solve_eigenproblem_legacy(
+        this, n_eval, eigenvalue, eigenvector);
+    }
+  }
+
+  //==================================================================
+  /// Solve the eigenproblem
+  //==================================================================
+  void Problem::solve_eigenproblem(
+    const unsigned& n_eval,
+    Vector<std::complex<double>>& alpha,
+    Vector<double>& beta,
+    Vector<Vector<std::complex<double>>>& eigenvector,
```

Comment:
Interface not suitable for distributed problems and therefore not suitable for implementation in the Problem class? ... or should there be a warning?

Reply:
I will make it work for distributed problems.

Reply:
Done.